src/Routes/Routes.tsx

```diff
@@ -18,12 +18,31 @@ function useLocationBackground() {
 
 export const Routes = () => {
   const { background, location } = useLocationBackground()
-  const { state } = useWallet()
+  const { connectDemo, state } = useWallet()
   const { appRoutes } = useBrowserRouter()
   const hasWallet = Boolean(state.walletInfo?.deviceId) || state.isLoadingLocalWallet
 
   return (
     <Switch location={background || location}>
+      <Route path='/demo'>
+        {() => {
+          const matchDemoPath = matchPath<{ appRoute: string }>(location.pathname, {
+            path: '/demo/:appRoute(.+)',
+          })
+          // Don't reconnect demo wallet if already connected
+          if (state.walletInfo?.deviceId !== 'DemoWallet') {
+            connectDemo()
```

Comment:
TIL that diffs are supported in comments, nice one @gomesalexandre.

Reply:
Saving a diff in clipboard with `git --no-pager diff | pbcopy` and applying it from clipboard with `pbpaste | git apply` are life-savers for sharing such solutions without needing to go with a new branch and commit for demoing it!

Reply:
Alrighty, so I did start coming across a bunch of weird behaviour and inconsistent state, particularly with the `isDemoWalletConnected` getting out of sync with the actual state, particularly when switching wallets. Instead, I opted for `shouldRedirectDemoRoute`, which only persists as true until the redirect executes.
While awaiting connectDemo did prevent the flag being updated before the wallet was connected, other side effects were still causing this to try and execute multiple times. By tracking the loading state, I was able to ensure the effect also early exits until we can actually redirect. With this change in place, we don't necessarily need to await the connection.
@gomesalexandre I would appreciate you thoughts on this approach. While everything seems to work as expected, I'm not sure if I've missed any significant performance implications resulting from these added side effects.

Reply:
Looks like a sound approach to me. `setShouldRedirectDemoRoute(false)` would potentially run too much, but in effect (pun intended), setting the exact same (primitive) state value as with previous state is effectively a no-op so that's fine. Tested it out and it works fine on initial and subsequential loads.